website/client/src/pages/settings/promoCode.vue

```diff
@@ -0,0 +1,138 @@
+  <div class="row standard-page">
+    <div class="col-12">
+      <h1
+        v-once
+        class="page-header"
+      >
+        {{ $t('promoCode') }}
+      </h1>
+
+      <div class="input-area">
+        <div
+          class="form-inline"
+          role="form"
+        >
+          <input
+            v-model="couponCode"
+            class="form-control"
+            type="text"
+            :placeholder="$t('promoPlaceholder')"
+          >
+        </div>
+        <div
+          v-once
+          class="small mt-2"
+        >
+          {{ $t('couponText') }}
+        </div>
+        <save-cancel-buttons
+          :hide-cancel="true"
+          primary-button-label="submit"
+          @saveClicked="enterCoupon()"
+        />
+        <div v-if="canCreateCoupons">
+          <hr>
+          <h4>{{ $t('generateCodes') }}</h4>
+          <div
+            class="form"
+            role="form"
+          >
+            <div class="form-group">
+              <input
+                v-model="codes.event"
+                class="form-control"
+                type="text"
+                placeholder="Event code (eg, 'wondercon')"
+              >
+            </div>
+            <div class="form-group">
+              <input
+                v-model="codes.count"
+                class="form-control"
+                type="number"
+                placeholder="Number of codes to generate (eg, 250)"
+              >
+            </div>
+            <div class="form-group">
+              <button
+                class="btn btn-primary"
+                type="submit"
+                @click="generateCodes(codes)"
+              >
+                {{ $t('generate') }}
+              </button>
+              <a
+                class="btn btn-secondary"
+                :href="getCodesUrl"
+              >{{ $t('getCodes') }}</a>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<style scoped>
+  .form-control {
+    width: 100%;
+  }
+</style>
+
+<script>
+import axios from 'axios';
+import { mapState } from '@/libs/store';
+import notifications from '@/mixins/notifications';
+import SaveCancelButtons from '@/pages/settings/components/saveCancelButtons.vue';
+
+export default {
+  components: { SaveCancelButtons },
+  mixins: [notifications],
+  data () {
+    return {
+      codes: {
+        event: '',
+        count: '',
+      },
+      couponCode: '',
+    };
+  },
+  computed: {
+    ...mapState({ user: 'user.data', credentials: 'credentials' }),
+    canCreateCoupons () {
+      return this.user.permissions.coupons;
+    },
+    getCodesUrl () {
+      if (!this.user) return '';
+      return '/api/v4/coupons';
+    },
+  },
+  mounted () {
+    this.$store.dispatch('common:setTitle', {
+      section: this.$t('settings'),
+      subSection: this.$t('promoCode'),
+    });
+  },
+  methods: {
+    generateCodes () {
+      // $http.post(ApiUrl.get() + '/api/v2/coupons/generate/
```

Comment:
So this is a null function?

If we want this to be functional, it should be an action in the client store that invokes `axios` to send the API request.

Reply:
Or at least an `axios` call here, if we don't care about putting it in the store.

Reply:
I think it's okay to remove it for now?

Reply:
removed it for now - once we need it we can just redo it in a better way